artifact-management/commands/setup-gem-environment.yml

```diff
@@ -4,10 +4,6 @@ parameters:
     description: The token to be used when fetching packages from Packagecloud.
     type: env_var_name
     default: PACKAGECLOUD_READ_TOKEN
-  gem_repo_host:
```

Comment:
Check me on this, but I *think* that we only use `gem_repo_host` and `gem_repo_key` for pushes, not for pulls. So either we need both of these parameters or we need neither of them.

But please check this, I could be wrong.

The same comment applies to [the place where you removed `gem_repo_host`](https://github.com/avvo/circleci-orbs/pull/6/files#diff-6e9e49cb9a90e06b4611ca23c6a0fd20L7) in the `gem-test` command.

Reply:
We're missing each other a bit on this one. Now that I've re-read my comments I can see why my point isn't clear. Let me try again.

I think that `bundle install` or `gem install` will _pull_ from artifactory without the `gem_repo_key` written to the `$HOME/.gem/credentials` file. I think that file is only used for _pushes_.

So in other words `setup-gem-environment` is configuring for two cases:

* One case where we are pulling from Packagecloud (credentials required) and Artifactory (credentials not required)
* One case where we are pushed to Packagecloud (credentials required) and Artifactory (credentials required)

We use `setup-gem-environment` in both cases, because it installs the maximum number of credentials required.

Is it necessary to separate this into `setup-gem-pull-environment` and `setup-gem-push-environment`? Probably not. It's correct but also somewhat pedantic.

I think that the removal of `gem_repo_host` without removing `gem_repo_key` - as you originally proposed - is probably the simplest way forward.

Reply:
I inclined to think that we should hear DRY philosophy for this case. Despite that setup-gem-environmen is configuring a case that is not required when pulling (Artifactory) I found better to have all “credential” logic in one single place.

Reply:
Other issue about this variables is:
But what I do not understand is why we are using this code like next:
```
- run:
    name: "Configure destination gem repo"
    command: |-
      printf -- "---\n:rubygems_api_key: Basic $AVVO_GEM_REPO_TOKEN\n" > $HOME/.gem/credentials
      chmod 0600 $HOME/.gem/credentials
```
	where there is no need to declare the parameter `gem_repo_key` (of type `env_var_name`) because is not used in the code.
	Instead of using a code more like suggested here: [https://circleci.com/docs/2.0/reusing-config/#environment-variable-name](Environment Variable Name) That it should look something like:
```
- run:
    name: "Configure destination gem repo"
    command: |-
      printf -- "---\n:rubygems_api_key: Basic **${<<parameters.gem_repo_key>>}**\n" > $HOME/.gem/credentials
      chmod 0600 $HOME/.gem/credentials
```
	Could the answer be something like?: Because we are not expecting the users to give a new value to this variables: we do not expect a use like:
```
jobs:
  test:
        docker:
      - image: circleci/ruby:2.6
    steps:
      - checkout
      - artifact-management/gem-test:
            gem_repo_key: other_value_for_this
```
	But if this is the case, why are we declaring the parameter `gem_repo_key`?

Reply:
We're declaring it for documentation, so that readers of the orb docs realize what effect the environment variable `AVVO_GEM_REPO_TOKEN` has on the commands that consume it.

We can implement the command script using either `$AVVO_GEM_REPO_TOKEN` or `<<parameters.gem_repo_key>>`; it works the same either way. However I don't think you can override an `env_var_name` parameter in an orb command call like you hypothesized in your example. So, just to reiterate, the purpose is documentation, not overridability.